src/components/Tables/Filters.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import { styled } from "styled-components";
+import { TableHeaders } from "@/components/Tables/Table";
+
+export interface Filter<Data, State> {
+    shouldFilter(data: Data, state: State): boolean;
```

Comment:
Ooo - haven't seen this syntax before - didn't know you can have functions in JS interfaces. I've always just done something like: `shouldFilter: (data: Data, state: State) => boolean;`
just curious - but is it common to do this in an interface? or would it be more of a class thing?